test/e2e/app-dir/metadata-streaming/metadata-streaming.test.ts

```diff
@@ -0,0 +1,83 @@
+import { nextTestSetup } from 'e2e-utils'
+import { retry, createMultiDomMatcher } from 'next-test-utils'
+
+describe('metadata-streaming', () => {
+  const { next } = nextTestSetup({
+    files: __dirname,
+  })
+
+  it('should load the page without delayed metadata', async () => {
+    const $ = await next.render$('/')
+    expect($('title').length).toBe(0)
+  })
+
+  it('should still load viewport meta tags even if metadata is delayed', async () => {
+    const $ = await next.render$('/slow')
+
+    expect($('meta[name="viewport"]').attr('content')).toBe(
+      'width=device-width, initial-scale=1'
+    )
+    expect($('meta[charset]').attr('charset')).toBe('utf-8')
+  })
+
+  it('should render the metadata in the browser', async () => {
+    const browser = await next.browser('/')
+    await retry(async () => {
+      expect(await browser.elementByCss('title').text()).toBe('index page')
+    })
+  })
+
+  it('should load the initial html without slow metadata during navigation', async () => {
+    // navigate from / to /slow, the metadata should be empty first, e.g. no title.
+    // then the metadata should be loaded after few seconds.
+    const browser = await next.browser('/')
+    await browser.eval(`document.getElementById('to-slow').click()`)
```

Comment:
why not `browser.elementById(...).click()`?

Reply:
damn, AI man, AI